homeassistant/components/tessie/sensor.py

```diff
@@ -0,0 +1,152 @@
+"""Sensor platform for Tessie integration."""
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    PERCENTAGE,
+    UnitOfEnergy,
+    UnitOfLength,
+    UnitOfPower,
+    UnitOfPressure,
+    UnitOfSpeed,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, TessieApi
+from .entity import TessieEntity
+
+PARALLEL_UPDATES = 0
+
+
+DESCRIPTIONS: dict[str, tuple[SensorEntityDescription, ...]] = {
+    TessieApi.CHARGE_STATE: (
+        SensorEntityDescription(
+            key="battery_level",
+            translation_key="battery_level",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=PERCENTAGE,
+            device_class=SensorDeviceClass.BATTERY,
+        ),
+        SensorEntityDescription(
+            key="battery_range",
+            translation_key="battery_range",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfLength.KILOMETERS,
+            device_class=SensorDeviceClass.DISTANCE,
+        ),
+        SensorEntityDescription(
+            key="charge_energy_added",
+            translation_key="charge_energy_added",
+            state_class=SensorStateClass.TOTAL_INCREASING,
+            native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+            device_class=SensorDeviceClass.ENERGY,
+        ),
+    ),
+    TessieApi.DRIVE_STATE: (
+        SensorEntityDescription(
+            key="speed",
+            translation_key="speed",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfSpeed.KILOMETERS_PER_HOUR,
+            device_class=SensorDeviceClass.SPEED,
+        ),
+        SensorEntityDescription(
+            key="power",
+            translation_key="power",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement=UnitOfPower.KILO_WATT,
+            device_class=SensorDeviceClass.POWER,
+        ),
+        SensorEntityDescription(
```

Comment:
Not needed - but I assume this is like normal car shift state, i.e. D, R, etc.

Could be worth settings options/ creating an enum to back it since it is finite.

Reply:
Didnt know this was even a thing, Added. Thanks!